react/features/notifications/components/native/Notification.tsx

```diff
@@ -0,0 +1,302 @@
+/* eslint-disable lines-around-comment */
+
+import React from 'react';
+import { WithTranslation } from 'react-i18next';
+import { Text, View } from 'react-native';
+
+import { IReduxState } from '../../../app/types';
+// @ts-ignore
+import { Avatar } from '../../../base/avatar';
+import { translate } from '../../../base/i18n/functions';
+import {
+    Icon,
+    IconCloseLarge,
+    IconInfoCircle,
+    IconUsers,
+    IconWarning
+    // @ts-ignore
+} from '../../../base/icons';
+import { getRemoteParticipants } from '../../../base/participants/functions';
+import { connect } from '../../../base/redux/functions';
+import { colors } from '../../../base/ui/Tokens';
+import BaseTheme from '../../../base/ui/components/BaseTheme.native';
+import Button from '../../../base/ui/components/native/Button';
+import IconButton from '../../../base/ui/components/native/IconButton';
+import { BUTTON_TYPES } from '../../../base/ui/constants.native';
+import { replaceNonUnicodeEmojis } from '../../../chat/functions';
+import { getKnockingParticipants } from '../../../lobby/functions';
+import { NOTIFICATION_ICON } from '../../constants';
+import AbstractNotification, {
+    type Props as AbstractNotificationProps
+    // @ts-ignore
+} from '../AbstractNotification';
+
+// @ts-ignore
+import styles from './styles';
+
+
+/**
+ * Default value for the maxLines prop.
+ *
+ * @type {number}
+ */
+const DEFAULT_MAX_LINES = 2;
+
+/**
+ * Secondary colors for notification icons.
+ *
+ * @type {{error, info, normal, success, warning}}
+ */
+
+const ICON_COLOR = {
+    error: colors.error06,
+    normal: colors.primary06,
+    success: colors.success05,
+    warning: colors.warning05
+};
+
+
+type Props = AbstractNotificationProps & WithTranslation & {
+    _participants: ArrayLike<any>;
+};
+
+
+/**
+ * Implements a React {@link Component} to display a notification.
+ *
+ * @augments Component
+ */
+class Notification extends AbstractNotification<Props> {
+
+    /**
+     * Creates action button configurations for the notification based on
+     * notification appearance.
+     *
+     * @private
+     * @returns {Object[]}
+     */
+    _mapAppearanceToButtons() {
+        const {
+            customActionHandler,
+            customActionNameKey
+            // @ts-ignore
+        } = this.props;
+
+        if (customActionNameKey?.length && customActionHandler?.length) {
+            return customActionNameKey?.map((customAction: string, index: number) => {
+                let buttonLabelStyle: Object | undefined;
+
+                if (customAction === 'lobby.reject') {
+                    buttonLabelStyle = styles.rejectBtnLabel;
```

Comment:
Why doesn't the web need this level of trickery? A generic notification component should not know about what its content is. Everything needs to come via props.

Reply:
Because web uses atlaskit.

Reply:
Ah bummer. Can you come up with a way to pass this as props please?

Reply:
Will tackle this deeper, yes.